SMLHelper/Patchers/CraftDataPatcher_Subnautica.cs

```diff
@@ -1,97 +1,301 @@
+            {
+                PatchUtils.PatchDictionary(CraftData.equipmentTypes, CustomEquipmentTypes);
+                EquipmentTypesAltered = CustomEquipmentTypes.Count;
+            }
+        }
+
+        private static void GetSlotTypePrefix()
+        {
+            if (CustomSlotTypes.Count > 0 && SlotTypesAltered < CustomSlotTypes.Count)
+            {
+                PatchUtils.PatchDictionary(CraftData.slotTypes, CustomSlotTypes);
+                SlotTypesAltered = CustomSlotTypes.Count;
+            }
+        }
+
+        private static void GetCraftTimePrefix()
+        {
+            if (CustomCraftingTimes.Count > 0 && CraftingTimesAltered < CustomCraftingTimes.Count)
+            {
+                PatchUtils.PatchDictionary(CraftData.craftingTimes, CustomCraftingTimes);
+                CraftingTimesAltered = CustomCraftingTimes.Count;
+            }
+        }
+
+        private static void GetCookedCreaturePrefix()
+        {
+            if (CustomCookedCreatureList.Count > 0 && CookedCreatureAltered < CustomCookedCreatureList.Count)
+            {
+                PatchUtils.PatchDictionary(CraftData.cookedCreatureList, CustomCookedCreatureList);
+                CookedCreatureAltered = CustomCookedCreatureList.Count;
+            }
+        }
+
+        private static void GetBackgroundTypesPrefix()
+        {
+            if (CustomBackgroundTypes.Count > 0 && BackgroundTypesAltered < CustomBackgroundTypes.Count)
+            {
+                PatchUtils.PatchDictionary(CraftData.backgroundTypes, CustomBackgroundTypes);
+                BackgroundTypesAltered = CustomBackgroundTypes.Count;
+            }
+        }
+
+        private static void GetBuildablePrefix()
+        {
+            if (CustomBuildables.Count > 0 && BuildablesAltered < CustomBuildables.Count)
+            {
+                PatchUtils.PatchList(CraftData.buildables, CustomBuildables);
+                BuildablesAltered = CustomBuildables.Count;
+            }
+        }
+
+        private static void AddCustomDataToOriginalDictionaries()
         {
             short added = 0;
             short replaced = 0;
             foreach (TechType techType in CustomTechData.Keys)
             {
                 ITechData smlTechData = CustomTechData[techType];
+                bool techExists = CraftData.techData.TryGetValue(techType, out CraftData.TechData techData);
+                bool sameData = false;
 
-                var techDataInstance = new CraftData.TechData
+                if (techExists)
                 {
-                    _techType = techType,
-                    _craftAmount = smlTechData.craftAmount
-                };
+                    sameData = smlTechData.craftAmount == techData.craftAmount &&
+                        smlTechData.ingredientCount == techData.ingredientCount &&
+                        smlTechData.linkedItemCount == techData.linkedItemCount;
```

Comment:
This isn't a good assumption to make.  
Just because the number of ingredients, linked items, and crafted amount are the same doesn't necessarily mean that the there wasn't a change.

Reply:
That's why if goes further and checks the ingredients and linked items one by one but only if that comes back true